docs/meeting-notes/foam-core-2020-07-11.md

```diff
@@ -0,0 +1,125 @@
+# Foam Core 2020-07-11
+
+Present: @jevakallio, @riccardoferretti
+
+### Tests
+
+- How do we know this approach works?
+  - Supports renaming
+  - Supports searching with (attribute-x)
+  - Find dead links
+
+### Getting started
+
+- Land work to master
+  - Create a foam-core package
+  -
+
+### Open questions
+
+- How should writing to files work
+  - What if affected notes have unsaved changes
+
+### Graph methods
+
+- get all
+- search by
+  - tag
+  - free text
+  - [[todo]]: how do vs code search editors work? are they pluggable? what do they need?
+- find dead links
+  - for linters
+- serialize/toJSON (for visualizers)
+- subscribe to changes
+
+### Node methods
+
+- rename
+- get links
+  - forward links (for link lists)
+  - backlinks (with surrounding context)
+
+### Node definition
+
+What do we need the node (and edge metadata) to contain:
+
+- `id`: tbd
+  - should be unique, needs some kind of unique gen function
+  - should be reconstructable even if links are not updated every time
+  - what happens during rename? is reparenting the graph going to be hard?
+  - do id's need to be persistent, or can we create them per in-memory session, keep them stable despite renames, and then next session generate a new id?
+  - Ideally should be a path to file, so it's easy to look up from the graph by id for renaming
+- `type`: Note | Image | etc
+- `title`: can be read from markdown title or frontmatter metadata
+- `path`: full path to file, relative to workspace (graph) root
+- `links`:
+  - `id`: File to link to
+  - `text`: The link label
+  - `type` markdown | mediawiki | image | http
+  - `section`
+  - `block` (ref)
+  - Positional data from AST?
+- `tags`
+
+### Markdown layer
+
+- `source`: raw markdown (rename?)
+- `ast`: raw markdown ast
+- `checksum`: if we do caching
+
+### Link text
+
+// some-file.md
+// # Some File
+
+Write -> Store on disk
+[[Some File]] -> [Some File](some-file.md)
+
+Editing
+[Some File](some-file.md)
+
+On disk (could be solved by migration)
+[[some-file]]
+[[Some File]]
+
+- docs/index.md -> Index
+- notes/index.md -> Index
+
+[[Index]]
+[[Index | notes/index.md]]
+
+[Index] docs/index.md
+[Index | notes/index.md]: notes/index.md
+
+[[Some File | path/to/some-file.md]]
+
+Do we apply any constraints:
+
+- `[[file-name-without-extension]]`
+- `[[file-name-with-extension.md]]`
+- `[[Title Cased File Name]]`
+
+Not supported by Markdown Notes:
+
+- `[[path/to/file-name.md]]` - Just use markdown links
```

Comment:
IMO this is absolutely needed (better yet in format of [[path/to/file-name]]). Foldering of notes would probably be pretty typical use case (is at least to me). It's not particularly hard to implement in comparison to flat notes, and looks like we are already building the base for functionality that's currently in Markdown Notes...

Reply:
Agreed